pages/index.js

```diff
@@ -3,16 +3,46 @@ import { vocs } from "../lib/db";
 import Card from "../components/Card";
 import styled from "styled-components";
 import Cardlist from "../components/Cardlist";
+import { useState, useEffect } from "react";
+
+function getXRandomIndexes(x, length) {
+  // x => how many cards you want
+  // length => how many cards there are
+
+  if (x > length) {
+    x = length;
+  }
+  const maxIndex = length - 1;
+  const arr = [];
+  while (arr.length < x) {
+    const randomNumber = Math.floor(Math.random() * (maxIndex + 1));
+    if (!arr.includes(randomNumber)) {
+      arr.push(randomNumber);
+    }
+  }
+  return arr;
+}
+
 export default function Home() {
+  const [cards, setCards] = useState(vocs);
+  const [indexes, setIndexes] = useState([]);
+  useEffect(() => {
+    setIndexes(getXRandomIndexes(5, cards.length));
+  }, []);
+
   return (
     <>
       <Header />
       <StyledMain>
-        <Cardlist>
-          {vocs.map((voc) => {
-            return <Card key={voc.id} voc={voc}></Card>;
-          })}
-        </Cardlist>
+        {cards.length > 0 ? (
+          <Cardlist>
+            {indexes.map((item) => {
```

Comment:
Another good point to think about. You are right, that was the reason why I did it. But after thinking about it there is no reason to not call it ``index``. Because the 'map-function-index' is only a parameter name (can call it 'broccoli' if I want) for the second parameter, called by the map-function (hope that was not so confusing).
So I changed it to ``index``